world-explore/src/app/home/home.page.ts

```diff
@@ -113,22 +98,29 @@ export class HomePage implements OnInit {
   }
 
   toggleFavorite(country: Country, event: any) {
-    const isStartSwipe = event.detail.side === "start";
-    const isNotFavorite = !this.favoriteCountries.has(country.country);
-    isStartSwipe && isNotFavorite ? this.favoriteCountries.add(country.country) : this.deleteCountry(country);
-    this.filterCountries();
+    if (event.detail.side === 'start') {
+      if (this.favoriteCountries.has(country.country)) {
+        this.favoriteCountries.delete(country.country);
+      } else {
+        this.favoriteCountries.add(country.country);
+      }
+      this.filterCountries(); // Update the filtered countries list
+    }
   }
 
   deleteCountry(country: Country) {
```

Comment:
using logical ( && )operators and ensure that the function only executes its operations when this.showFavorites is true